### PR TITLE
NPE fix in DataResultViewerTable (5246)

### PR DESCRIPTION
**Pull request by eugene7646**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java [hunk 1]
```diff
@@ -736,16 +738,45 @@ private synchronized void loadColumnVisibility() {
...
             }
         }
 
-        return new ArrayList<>(propertiesMap.values());
+        // if there are no discontinuities, we are done
+        if (availablePositions.isEmpty()) {
+            return;
+        }
+
+        // otherwise, move map the elemets into the available positions.
```
Comment (APriestman) [inline Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: some typos in this sentence

File: Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java [hunk 2]
```diff
@@ -736,16 +738,45 @@ private synchronized void loadColumnVisibility() {
             }
         }
 
-        // If none of the properties had previous settings, we should decrement
-        // each value by the number of properties to make the values 0-indexed.
-        if (noPreviousSettings) {
```
Comment (APriestman) [inline Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: noPreviousSettings is no longer used so you could remove it.

**Comment by eugene7646** [comment]
@APriestman I have updated the PR

File: Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java [hunk 3]
```diff
@@ -768,7 +767,7 @@ private void compactPropertiesMap() {
             return;
         }
 
-        // otherwise, move map the elemets into the available positions. 
+        // otherwise, move map elemets into the available positions.
```
Comment (APriestman) [inline Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: "elements" is still misspelled

Comment (eugene7646) [reply Core/src/org/sleuthkit/autopsy/corecomponents/DataResultViewerTable.java]: @APriestman I guess I'm capable of multiple misspellings in one line of text! Fixed. I hope...

**Pull request merged**